apps/api/src/domain/domainModelValidators/spatialFeatureValidator/geometricFeatureValidator/buildGeometricFeatureValidatorChain/buildPointChainValidator.ts

```diff
@@ -0,0 +1,52 @@
+import { GeometricFeature } from 'apps/api/src/domain/models/spatial-feature/GeometricFeature';
+import { PointCoordinates } from 'apps/api/src/domain/models/spatial-feature/types/Coordinates/PointCoordinates';
+import { GeometricFeatureType } from 'apps/api/src/domain/models/spatial-feature/types/GeometricFeatureType';
+import { InternalError } from 'apps/api/src/lib/errors/InternalError';
+import { isNumber } from 'class-validator';
+
+const pointChainValidator = (
+    input: GeometricFeature,
+    incomingErrors: InternalError[]
+): InternalError[] => {
+    const { type, coordinates } = input;
+
+    // Pass on the responsibility
+    if (type !== GeometricFeatureType.point) return incomingErrors;
+
+    // Validate Coordinates
+    const allErrors: InternalError[] = [];
+
+    if (coordinates.length !== 2)
+        allErrors.push(
+            new InternalError(
+                `A Geometric Point must have 2 coordinates. Received: ${coordinates.length}`
+            )
+        );
+
+    const coordinateTypeErrors = (coordinates as PointCoordinates).reduce(
+        (accumulatedErrors: InternalError[], coordinate, index) => {
+            if (
+                !isNumber(coordinate, {
+                    allowInfinity: false,
```

Comment:
Are these class-validator options?

Reply:
Yes. I am not sure how I feel about using class-validator methods for domain model validation. We should think about this. Maybe we need our own custom validator library, which itself can use class-validator if it sees fit?

Reply:
I took a step in this direction.